platforms/jvm/plugins-java-base/src/main/java/org/gradle/api/plugins/JavaBasePlugin.java

```diff
@@ -80,11 +79,23 @@
 /**
  * <p>A {@link org.gradle.api.Plugin} which compiles and tests Java source, and assembles it into a JAR file.</p>
  *
+ * Note that this plugin is automatically applied to most projects that build any JVM language source.  It would be
+ * more appropriately titled the {@code jvm-base} plugin - as it creates the {@link SourceSet}s used by Java, Groovy and Kotlin
+ * plugins.
+ *
+ * The {@link JavaPluginExtension} extension created by this plugin would similarly be more appropriately named the {@code jvm}
+ * extension instead of the current name: {@code java}.  It is used to configure all {@link org.gradle.jvm.component.internal.JvmSoftwareComponentInternal JvmSoftwareComponentInternal}
+ * components.  At present there is only one such component - the {@code java} component added by the
+ * {@link org.gradle.api.plugins.JavaPlugin JavaPlugin} - but multiple components of this type may be created by
+ * these and other plugins in the future.
```

Comment:
I think this is kind of confusing. This plugin doesn't create SourceSets, it configures their conventions if they exist.

I don't think we should say "it would make more sense if this was called something else" in a public doc.

I think it is OK to say it create an extension called `java` with type `JavaPluginExtension`